### ISSUE TITLE
Server-side Recording time stamp and the Unix time stamp in a recording status Rest API response is differ by 2-3 ms

### ISSUE THREAD

**Issue by yashtandon113** [ant-media/Ant-Media-Server#4387]
### Short description
Date and time in Recording Name on server side after the recording is finished.
2022/08/31 10:26:02 [info] [ams] 10021237-2022-08-31_06-26-02.**459**.mp4

Rest API response while enabling the recording.
2022/08/31 10:26:02 [info] [ams] '{\"success\":true,\"message\":\"1661941562461\",\"dataId\":null,\"errorId\":0}'

Time Stamp (1661941562461) --> Wednesday, August 31, 2022 10:26:02.**461** AM

There is a 2-3 ms gap, therefore it is impossible to target the recordings precisely.

### Environment
* Ant Media Server version: Tested with v2.4.3 & above

### Steps to reproduce
1. Enable the option in settings "Add Date-Time to Record File names"
2. Publish the stream
3. Call the [recording status](https://antmedia.io/rest/#/BroadcastRestService/enableRecording) Rest API
4. Stop the recording
5. Compare the time stamp in the recording name on the server side and the time stamp in API response

### Expected behavior
Both the server side and the API response should have the same recording timestamp.

### Actual behavior
The time stamps differ by 2-3 ms in multiple tests.

### Recommended solution by user
Can we maybe use the ‘dataId’ field? If we could have that field return the exact recording name on the start recording Rest api response, I think that would be a good solution as well.

**Comment by danavramescu** [comment]
Possible solutions:
1. Use dataId field to return the actual recording name, in which case we no longer have to look at the timestamp (message field) but just grab the recording name as received in the dataId field

2 .Use the response 'timestamp' as recording name suffix ( of course, after the timestamp ms difference issue is fixed)
eg instead of 10021237-2022-08-31_06-26-02.459.mp4 
10021237_1661941562461.mp4 
Btw, isn't this a generally better solution overall? to just use the timestamp? why do you even bother creating dates suffixes in GMT (where summertime/wintertime creates a 1h offset), and not just use timestamps suffixes? since those are unique anyway.

**Comment by danavramescu** [comment]
Or, as a third option, allow users to create their own custom names for those recordings. 
F.e when i call the startRecording restApi, we could pass a custom name we desire, and the recording would be created using that exact name. But not mandatory to do so if the timestamp ms issue is solved. Just thought maybe this would be a viable solution aswell

**Comment by SelimEmre** [comment]
Hi @danavramescu,

Have a good day. Thank you for these options. I can recommend you to use `creationDate`, `startTime` and `duration` parameters instead of using filename. We are providing these details on VoD REST API. Here is the example of JSON data:
```
[
{
"streamName": "stream1_2.mp4",
"vodName": "stream1_2.mp4",
"streamId": "stream1",
"creationDate": 1662970017959,
"startTime": 1662969872643,
"duration": 144777,
"fileSize": 17883671,
"filePath": "streams/stream1_2.mp4",
"vodId": "247385866269546809393846",
"type": "streamVod",
"previewFilePath": null
}
]
```

**Comment by danavramescu** [comment]
> Hi @danavramescu,
> 
> Have a good day. Thank you for these options. I can recommend you to use `creationDate`, `startTime` and `duration` parameters instead of using filename. We are providing these details on [VoD REST API](https://antmedia.io/rest/#/VoD%20Rest%20Service/getVoD). Here is an example of JSON data:
> 
> ```
> [
> {
> "streamName": "stream1_2.mp4",
> "vodName": "stream1_2.mp4",
> "streamId": "stream1",
> "creationDate": 1662970017959,
> "startTime": 1662969872643,
> "duration": 144777,
> "fileSize": 17883671,
> "filePath": "streams/stream1_2.mp4",
> "vodId": "247385866269546809393846",
> "type": "streamVod",
> "previewFilePath": null
> }
> ]
> ```

Hi!

at the moment we are facing another issue with mp4 recs not starting at all, not even from the ams admin. 
[root@slave SecretFriends]# curl -X PUT "https://ant.imcdev.com:5443/webrtcappEEDEV/rest/v2/broadcasts/10021237/recording/true?recordType=mp4" -H "accept: application/json"
{"success":false,"message":"mp4 recording couldn't be started","dataId":null,"errorId":0}

so first we have to figure out what changed and why this isn't working anymore. 

Afterwards we can continue looking into what you advised.
But before getting there i want to mention this:
FFMPEG will only work with exact filenames. so what we need is for the response to give us something (eg dataId could be the vodId) so that we can retrieve the filename (filePath) from via restApi). In all possible scenarios, what we need is to have a quick and 100% accurate way of grabbing the target recording filename, exactly as it gets created)

So please advise the correct steps to take in order to target the correct recording and grab it's filename
1. we start the recording like this
$startRecCmd = 'curl -X PUT "https://ant.imcdev.com:5443/webrtcappEEDEV/rest/v2/broadcasts/'.$this->id.'/recording/true?recordType=mp4" -H "accept: application/json"';

2.
Now, from older logs, i see a successful response message doesn't contain too much info, except that timestamp which is off by miliseconds (when comparing to the actual date appended to the recording filename--something we are now trying to get around and use restApi instead)
2022/08/31 07:17:44 [info] [ams] '{\"success\":true,\"message\":\"1661930264637\",\"dataId\":null,\"errorId\":0}'

So, what can we do with the info we have in this response to call the restapi and manage to find the corresponding recording? Maybe you can provide an example of how to use the 'message' field in order to find the recording's filename?

**Comment by SelimEmre** [comment]
Hi @danavramescu,

I was on annual vacation. I couldn't reply to you. We just implemented VoD ID to dataId on the latest snapshot. You can download it to your account. 

> what we need is to have a quick and 100% accurate way of grabbing the target recording filename, exactly as it gets created)
We have Webhooks for VoD finish event. Did you check that? Maybe it can help your scenario: https://resources.antmedia.io/docs/using-webhooks#webhooks-list

I also added a fix regarding your request. It's in PR right now. I jıst wanted to let you know: https://github.com/ant-media/Ant-Media-Server/pull/4452

Best Regards,
Selim

**Comment by mekya** [comment]
This fix is available in the latest version(2.5.0)

FYI @danavramescu
